board/arm-64bit-toolchain.cmake

```diff
@@ -0,0 +1,63 @@
+## This file handles cross-compilation configurations for aarch64,
+## known as arm64. The objective of this file is to find and assign
+## cross-compiler and the entire toolchain.
+## It works best with buildroot toolchain, when using it the user
+## needs to set the: TOOLCHAIN_PREFIX and CMAKE_SYSROOT from the
+## command line.
+
+set(CMAKE_SYSTEM_NAME Linux)
+set(CMAKE_SYSTEM_PROCESSOR ARM)
+
+option(BUILDROOT "Use the buildroot toolchain to cross-compile." OFF)
+option(LINARO "Use the linaro toolchain to cross-compile." OFF)
+
+if(BUILDROOT)
+  set(CMAKE_SYSROOT ${TOOLCHAIN_PATH}/aarch64-buildroot-linux-gnu/sysroot)
+  set(TOOLCHAIN_PREFIX ${TOOLCHAIN_PATH}/bin/aarch64-buildroot-linux-gnu-)
+  message(STATUS "TOOLCHAIN_PREFIX is defined as: ${TOOLCHAIN_PREFIX}")
+  message(STATUS "CMAKE_SYSROOT is defined as: ${CMAKE_SYSROOT}")
+elseif(LINARO)
+  set(CMAKE_SYSROOT ${TOOLCHAIN_PATH}/aarch64-linux-gnu)
+  set(TOOLCHAIN_PREFIX ${TOOLCHAIN_PATH}/bin/aarch64-linux-gnu-)
+  message(STATUS "TOOLCHAIN_PREFIX is defined as: ${TOOLCHAIN_PREFIX}")
+  message(STATUS "CMAKE_SYSROOT is defined as: ${CMAKE_SYSROOT}")
+else ()
```

Comment:
@rcurtin, that what I came with after hours of thinking and trying, cmake is not suitable to manipulate strings. thus we need to use boolean.
If users set `BUILDROOT` or `TOOLCHAIN_PATH`, there will be an error.
This model will allow us to support more toolchains in the future, as users might needs to